nx/lib/nx.ex

```diff
@@ -1195,15 +1182,141 @@ defmodule Nx do
       Use offset > 0 for diagonals above the main diagonal,
       and offset < 0 for diagonals below the main diagonal.
 
+  ## Error cases
+
+      iex> Nx.take_diagonal(Nx.tensor([0, 1, 2]))
+      ** (ArgumentError) take_diagonal/2 expects tensor of rank 2, got tensor of rank: 1
+
+      iex> Nx.take_diagonal(Nx.iota({3, 3}), offset: 3)
+      ** (ArgumentError) offset must be less than breadth when positive, got: 3
+
+      iex> Nx.take_diagonal(Nx.iota({3, 3}), offset: -4)
+      ** (ArgumentError) absolute value of offset must be less than length when negative, got: -4
```

Comment:
I don't think length and breadth are names we use "publicly". WDYT about using "must be less then length of axis 0" and "must be less than length of axis 1" instead?

Reply:
> I don't think length and breadth are names we use "publicly". 

You're right!

> WDYT about using "must be less then length of axis 0" and "must be less than length of axis 1" instead?

I think this is probably the best we can do.